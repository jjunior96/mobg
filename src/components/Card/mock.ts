import { v4 } from 'uuid';

import { CardProps } from '.';

export const CardMock: CardProps[] = [
  {
    id: v4(),
    type: 'house',
    location: 'Avenida Brasil',
    price: '1.160',
    meters: 64,
    rooms: 2,
    garage: 2,
    img: '/img/house-1.jpeg'
  },
  {
    id: v4(),
    type: 'apartment',
    location: 'Avenida Uruguai',
    price: '1.000',
    meters: 58,
    rooms: 2,
    garage: 1,
    img: '/img/house-1.jpeg'
  },
  {
    id: v4(),
    type: 'house',
    location: 'Avenida Brasil',
    price: '2.310',
    meters: 140,
    rooms: 4,
    garage: 4,
    img: '/img/house-1.jpeg'
  }
];
