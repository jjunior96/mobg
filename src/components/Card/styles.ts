import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    border: 0.2rem solid ${theme.colors.gray_100};

    border-radius: ${theme.border.radius};

    width: 100%;
    max-width: 34rem;

    overflow: hidden;
  `}
`;

export const Content = styled.div``;

export const ImgContainer = styled.img`
  width: 100%;
  object-fit: cover;
`;

export const InfoContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xsmall};
  `}
`;

export const InfoFooter = styled.footer`
  ${({ theme }) => css`
    width: 100%;

    margin-top: ${theme.spacings.xsmall};
  `}
`;

export const Type = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.blue_200};
    text-transform: capitalize;

    font-size: ${theme.font.sizes.xsmall};

    margin-bottom: ${theme.spacings.xxsmall};
  `}
`;

export const Price = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.blue_200};

    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};

    margin-bottom: ${theme.spacings.xsmall};

    strong {
      color: ${theme.colors.blue_300};
    }
  `}
`;

export const Location = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.blue_200};

    font-size: ${theme.font.sizes.xsmall};
  `}
`;

export const ItemContainer = styled.ul`
  width: 100%;

  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const ItemList = styled.li`
  ${({ theme }) => css`
    list-style: none;

    width: 100%;
    height: 100%;

    display: flex;
    flex-direction: column;

    color: ${theme.colors.blue_200};

    div {
      display: flex;
      flex-direction: column;
    }

    svg {
      color: ${theme.colors.primary};
    }

    p {
      font-size: ${theme.font.sizes.xsmall};
    }

    & + li {
      border-left: 1px solid ${theme.colors.gray_100};

      padding-left: ${theme.spacings.xsmall};
    }
  `}
`;
