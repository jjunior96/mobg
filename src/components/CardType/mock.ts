import { v4 } from 'uuid';

import { CardTypeProps } from '.';

export const CardTypeMock: CardTypeProps[] = [
  {
    id: v4(),
    type: 'Apartamentos',
    description: 'Lofts, studios, kitnets...',
    amount: 126,
    link: '#',
    img: '/img/house-1.jpeg'
  }
];
