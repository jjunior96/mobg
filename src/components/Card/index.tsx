import {
  BiRuler as MetersIcon,
  BiBed as RoomsIcon,
  BiCar as GarageIcon
} from 'react-icons/bi';

import * as S from './styles';

export type CardProps = {
  id: string;
  type: 'house' | 'apartment';
  img: string;
  price: string;
  location: string;
  meters: number;
  rooms: number;
  garage: number;
};

const Card = ({
  type,
  img,
  price,
  location,
  meters,
  rooms,
  garage
}: CardProps) => {
  return (
    <S.Container>
      <S.Content>
        <S.ImgContainer src={img} alt={`Foto da ${location}`} />

        <S.InfoContainer>
          <S.Type>{type}</S.Type>

          <S.Price>
            R$ <strong>{price}</strong>,00
          </S.Price>

          <S.Location>{location}</S.Location>

          <S.InfoFooter>
            <S.ItemContainer>
              <S.ItemList>
                <div>
                  <MetersIcon />
                  <p>{meters}m2</p>
                </div>
              </S.ItemList>
              <S.ItemList>
                <div>
                  <RoomsIcon />
                  <p>{rooms} Quartos</p>
                </div>
              </S.ItemList>
              <S.ItemList>
                <div>
                  <GarageIcon />
                  <p>{garage} Vaga</p>
                </div>
              </S.ItemList>
            </S.ItemContainer>
          </S.InfoFooter>
        </S.InfoContainer>
      </S.Content>
    </S.Container>
  );
};

export default Card;
