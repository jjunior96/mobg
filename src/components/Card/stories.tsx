import { Story, Meta } from '@storybook/react/types-6-0';

import Card, { CardProps } from '.';
import { CardMock } from './mock';

export default {
  title: 'Card',
  component: Card,
  args: CardMock[0]
} as Meta;

export const Default: Story<CardProps> = (args) => <Card {...args} />;
