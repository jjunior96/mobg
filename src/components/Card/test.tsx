import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../utils/tests/helpers';

import Card from '.';
import { CardMock } from './mock';

const props = CardMock[0];

describe('<Card />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Card {...props} />);

    expect(screen.getByRole('heading', { name: /Card/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
