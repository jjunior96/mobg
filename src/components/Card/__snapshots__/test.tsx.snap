// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the heading 1`] = `
.c0 {
  border: 0.2rem solid #e6e6eb;
  border-radius: 0.8rem;
  width: 100%;
  max-width: 34rem;
  overflow: hidden;
}

.c1 {
  width: 100%;
  object-fit: cover;
}

.c2 {
  width: 100%;
  padding: 1.6rem;
}

.c6 {
  width: 100%;
  margin-top: 1.6rem;
}

.c3 {
  color: #5d5e7a;
  text-transform: capitalize;
  font-size: 1.6rem;
  margin-bottom: 1rem;
}

.c4 {
  color: #5d5e7a;
  font-size: 2.4rem;
  font-weight: 600;
  margin-bottom: 1.6rem;
}

.c4 strong {
  color: #292b52;
}

.c5 {
  color: #5d5e7a;
  font-size: 1.6rem;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  list-style: none;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #5d5e7a;
}

.c8 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 svg {
  color: #5c63ef;
}

.c8 p {
  font-size: 1.6rem;
}

.c8 + li {
  border-left: 1px solid #e6e6eb;
  padding-left: 1.6rem;
}

<div
  class="c0"
>
  <div
    class=""
  >
    <img
      alt="Foto da Avenida Brasil"
      class="c1"
      src="/img/house-1.jpeg"
    />
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        house
      </p>
      <p
        class="c4"
      >
        R$ 
        <strong>
          1.160
        </strong>
        ,00
      </p>
      <p
        class="c5"
      >
        Avenida Brasil
      </p>
      <footer
        class="c6"
      >
        <ul
          class="c7"
        >
          <li
            class="c8"
          >
            <div>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.875 7H3.125C1.953 7 1 7.897 1 9v6c0 1.103.953 2 2.125 2h17.75C22.047 17 23 16.103 23 15V9c0-1.103-.953-2-2.125-2zm0 8H3.125c-.057 0-.096-.016-.113-.016-.007 0-.011.002-.012.008l-.012-5.946c.007-.01.052-.046.137-.046H5v3h2V9h2v4h2V9h2v3h2V9h2v4h2V9h1.875c.079.001.122.028.125.008l.012 5.946c-.007.01-.052.046-.137.046z"
                />
              </svg>
              <p>
                64
                m2
              </p>
            </div>
          </li>
          <li
            class="c8"
          >
            <div>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 9.557V3h-2v2H6V3H4v6.557C2.81 10.25 2 11.525 2 13v4a1 1 0 0 0 1 1h1v4h2v-4h12v4h2v-4h1a1 1 0 0 0 1-1v-4c0-1.475-.811-2.75-2-3.443zM18 7v2h-5V7h5zM6 7h5v2H6V7zm14 9H4v-3c0-1.103.897-2 2-2h12c1.103 0 2 .897 2 2v3z"
                />
              </svg>
              <p>
                2
                 Quartos
              </p>
            </div>
          </li>
          <li
            class="c8"
          >
            <div>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m20.772 10.156-1.368-4.105A2.995 2.995 0 0 0 16.559 4H7.441a2.995 2.995 0 0 0-2.845 2.051l-1.368 4.105A2.003 2.003 0 0 0 2 12v5c0 .753.423 1.402 1.039 1.743-.013.066-.039.126-.039.195V21a1 1 0 0 0 1 1h1a1 1 0 0 0 1-1v-2h12v2a1 1 0 0 0 1 1h1a1 1 0 0 0 1-1v-2.062c0-.069-.026-.13-.039-.195A1.993 1.993 0 0 0 22 17v-5c0-.829-.508-1.541-1.228-1.844zM4 17v-5h16l.002 5H4zM7.441 6h9.117c.431 0 .813.274.949.684L18.613 10H5.387l1.105-3.316A1 1 0 0 1 7.441 6z"
                />
                <circle
                  cx="6.5"
                  cy="14.5"
                  r="1.5"
                />
                <circle
                  cx="17.5"
                  cy="14.5"
                  r="1.5"
                />
              </svg>
              <p>
                2
                 Vaga
              </p>
            </div>
          </li>
        </ul>
      </footer>
    </div>
  </div>
</div>
`;
